src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java

```diff
@@ -2042,12 +2040,8 @@ default MessagePaginationAction getIterableHistory()
     @CheckReturnValue
     default RestAction<Void> removeReactionById(@Nonnull String messageId, @Nonnull String unicode)
     {
-        Checks.isSnowflake(messageId, "Message ID");
-        Checks.noWhitespace(unicode, "Emoji");
-
-        final String code = EncodingUtil.encodeUTF8(unicode);
-        final Route.CompiledRoute route = Route.Messages.REMOVE_REACTION.compile(getId(), messageId, code, "@me");
-        return new RestActionImpl<>(getJDA(), route);
+        // removes code redundancy but kind of bad design
+        return ((TextChannel) this).removeReactionById(messageId, unicode, getJDA().getSelfUser());
```

Comment:
What if this is not a `TextChannel`?

Reply:
Well. I thought it has to be castable since TextChannel implements MessageChannel. Isn't this the case?

I know that it was at least bad design but I couldn't come up with a better approach, do you have any ideas how to get rid of the code redundancy here?

Reply:
You can't cast a `PrivateChannel` to a `TextChannel` they both implement this interface.

Reply:
Just tried it myself really quick.
```
Exception in thread "main" In MessageChannel
java.lang.ClassCastException: implementscheck.PrivateChannel cannot be cast to implementscheck.TextChannel
```
You're totally right ofc. Somehow I thought that only happens if you try to cast something like a String to a Double (nothing in common like an abstract class or an interface). Somehow I thought it would be okay since both classes are implementing MessageChannel. :man_facepalming:

The problem is: The method in the TextChannel can also be with another user so simply calling MessageChannel's default method won't work. Also no private helping functions are allowed in an Interface. Guess it has to be implemented twice then, hasn't it?